### ISSUE TITLE
feat: add event for clicking on searchbar search icon

### ISSUE THREAD

**Issue by Monomachus** [ionic-team/ionic#19997]
# Feature Request
I'd like to have a possibility to assign an event listener to searchbar icon

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

I have the angular (search) eventListener, but it would only be firing when we click enter on the ion-searchbar. When we click on the search icon nothing happens. What I would like to happen is actually the same search action 

**Describe Preferred Solution**

The easiest way would be to have defined a search event and that event would apply on enter and on clicking the search icon. 

**Describe Alternatives**

Another one way would be just allowing to define an event when clicking the icon. It that case we can still use (search) from search bar, but also an event listener for the search icon.

**Related Code**

**Additional Context**

I've seen this solution https://forum.ionicframework.com/t/ionic-search-bar-search-icon-clicked/77072/4 which implies using elementRef, but not sure if this is a great solution because of the fact that those listeners could leak if not removed, etc.

Also I've read about HostListener, but when I add it to the searchbar and set click event, it is always showing as event target the searchbar (even when I click on the icon itself)

**Comment by liamdebeasi** [comment]
Thanks for the feature request. Can you explain your use case for this feature a bit more?

**Comment by Monomachus** [comment]
Ok, so in my case, a user is typing in a search bar and the search shouldn't happen once the user ends typing, but instead, when he/she press the enter button on the keyboard or which seems more obvious (at least in my case) - press the search icon. 

![image](https://user-images.githubusercontent.com/65164/69705807-f4061b80-10fe-11ea-8f60-b08e1356edb2.png)

When I inspected the code I saw that 'clear' icon is, in fact, a button.
So it would be great if the search icon could also be pressed, and once it's pressed I'd have an event handler to pass a method (what should be performed when the icon is searched).
In my case only when the icon would be pressed I'd do the call to the API, and show the list of results.

**Comment by AbhishekJ01** [comment]
Is there any update on how to achieve this?? As I have same exact requirement on my project.

**Comment by messkan** [comment]
any updates please?

**Comment by PulpFuture** [comment]
Is anyone aware of a workaround?

**Comment by Monomachus** [comment]
I've create a new component with a searchbar and a button :( 
Not ideal, but it works as asked by the end users

![image](https://user-images.githubusercontent.com/65164/89552206-85906600-d814-11ea-8a56-16f057730873.png)

**Comment by PulpFuture** [comment]
> I've created a new component with a search bar and a button :(
> Not ideal, but it works as asked by the end-users
> 
> ![image](https://user-images.githubusercontent.com/65164/89552206-85906600-d814-11ea-8a56-16f057730873.png)

Yeah I just hid the icon and put a button in the search bar for now. I couldn't get query selectors or local references to pick it up to add a click.
